src/crypto/sha/sha512.cpp

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/crypto/sha/sha512.hpp>
+
+#include <openssl/sha.h>
+
+namespace libp2p::crypto {
+  common::Hash512 sha512(std::string_view input) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    const auto *bytes_ptr = reinterpret_cast<const uint8_t *>(input.data());
```

Comment:
if possible, pls avoid using reinterpret_cast. Can we change method signature for this aim?

Reply:
please rewrite it using 
```std::vector<const uint8_t> bytes{input.begin(), input.end()};
    return sha256(bytes);```
both here and in `sha256`